internal/buffer/save.go

```diff
@@ -150,6 +150,23 @@ func (b *Buffer) saveToFile(filename string, withSudo bool, autoSave bool) error
 	// Removes any tilde and replaces with the absolute path to home
 	absFilename, _ := util.ReplaceHome(filename)
 
+	// Resolve a possible symlink to his absolute path
+	fileInfo, err := os.Lstat(absFilename)
+	if !os.IsNotExist(err) {
+		if err != nil {
+			return err
+		}
+		if fileInfo.Mode()&os.ModeSymlink != 0 {
+			dstFilename, err := os.Readlink(absFilename)
```

Comment:
What if `dstFilename` is a symlink too?

Reply:
Hm, right...chained symlinks should be supported too.
We can iter over this section, but need to check for recursive links again. :thinking:

Reply:
Ok, it seems like recursion is already covered by the standard functions...

```shell
lrwxrwxrwx  1 user user       3 29. Apr 20:40 ln1 -> ln1
```

```shell
lrwxrwxrwx  1 user user       3 29. Apr 20:42 ln1 -> ln2
lrwxrwxrwx  1 user user       3 29. Apr 20:42 ln2 -> ln1
```

...because both lead to:

```shell
$ micro ln1 
stat ln1: too many levels of symbolic links

Press enter to continue
```

Reply:
Now run `micro ln1` first, *before* creating these 2 symlinks pointing to each other, then create them in a parallel terminal, and then try to save the file in micro.

Reply:
Damn, I hate it when you proof me wrong in just a few seconds. :wink: